internal/download/download.go

```diff
@@ -115,50 +118,116 @@ func DownloadFile(url, outPath, header, token string) error {
 	return nil
 }
 
+func documentHashKey(document workflow.SpeakeasyRegistryDocument) string {
+	builder := strings.Builder{}
+	builder.WriteString(document.WorkspaceSlug)
+	builder.WriteString(":")
+	builder.WriteString(document.OrganizationSlug)
+	builder.WriteString(":")
+	builder.WriteString(document.NamespaceName)
+	builder.WriteString(":")
+	builder.WriteString(document.NamespaceID)
+	builder.WriteString(":")
+	builder.WriteString(document.Reference)
+	return builder.String()
```

Comment:
nit: this would be much more readable with `fmt.Sprintf` instead of a builder

Reply:
Probably would; I've been using builders though everywhere since we managed to squeeze out a 20% speed improvement in CLI through swapping out a few `fmt.fSprintf` in hot paths to string builders. Not worth it here, but it has been habit-forming.